### PR TITLE
Secp256k1

### PR DESCRIPTION
**Pull request by subtly**

### CODE REVIEW

File: libdevcrypto/Common.cpp [hunk 1]
```diff
@@ -52,21 +62,43 @@ Address dev::ZeroAddress = Address();
...
-	s_secp256k1.toPublic(_secret, p);
-	return s_secp256k1.toAddress(p);
+	return move(toAddress(toPublic(_secret)));
+}
+
+Public dev::recover(Signature const& _sig, h256 const& _message)
+{
+	bytes o(65);
+	int pubkeylen;
+	return secp256k1_ecdsa_recover_compact(_message.data(), _sig.data(), o.data(), &pubkeylen, 0, _sig[64]) ? move(*(Public*)(o.data()+1)) : Public();
```
Comment (chriseth) [inline libdevcrypto/Common.cpp]: spaces

File: secp256k1/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+language: c
```
Comment (chriseth) [inline secp256k1/.travis.yml]: Is that used?

File: secp256k1/.gitignore [hunk 3]
```diff
@@ -0,0 +1,36 @@
+bench_inv
```
Comment (chriseth) [inline secp256k1/.gitignore]: Are these needed? Perhaps better to add some to `/.gitignore`?

**Comment by subtly** [comment]
This version of libsecp256k1 (github) isn't for production. Will open new PR.

**Pull request closed without merging**